network/src/main/java/org/jboss/as/network/OutboundSocketBinding.java

```diff
@@ -215,21 +215,21 @@ public InetAddress getOptionalSourceAddress() {
      * this outbound socket binding has a port offset. To get the absolute source port, use the {@link #getAbsoluteSourcePort()}
      * method.
      *
-     * @return the source port for this outbound socket binding
+     * @return the source port for this outbound socket binding not accounting for port offset/fixation; {@code null} if an ephemeral port should be used
      */
     public Integer getSourcePort() {
-        return this.sourcePort;
+        return sourcePort == 0 ? null : this.sourcePort;
```

Comment:
Looks like a null check on sourcePort is missing here because sourcePort is Integer and therefore can be null. If sourcePort is null then the comparison "== 0" will cause problems when trying to debox sourcePort to int.

Just tried the following similar code and it crashed with NPE:
`Integer sourcePort = null;`
`int x = (sourcePort == 0 ? null : sourcePort);`